language-design.md

```diff
@@ -28,7 +103,54 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
-
+Guy Steele gives us several traits of a good language in "Growing a Language".
+According to Steele, the size of the language is one important indicator. As he
```

Comment:
Good point. Honestly, I'm not really sure how to answer this. I'm tempted to say that with enough experience a programmer should be able to tell when their language is too small or too large, even if they can't articulate why. But that's not a useful answer. A possible indicator of a language being too large might be that new users have a hard time learning all of it, but there are so many confounding factors when it comes to speed of language-learning. For too-small languages, maybe we could look for something Steele mentioned: does the user have to implement many functions that feel like they should be primitives? That's already kind of nebulous, though. We wouldn't say ContextFree is too small because someone trying to do data analysis has to do a lot of overhead first, because ContextFree isn't concerned with data analysis. Even for a general purpose language, it's hard to tell when something should be added. The many types of numbers mentioned by Steele may feel like they should be primitives to someone working with numbers, but that doesn't mean they all belong in the language.

tl;dr I'm not really sure